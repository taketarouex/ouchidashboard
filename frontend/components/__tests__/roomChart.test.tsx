import React from 'react';
import { render, waitForElementToBeRemoved } from '@testing-library/react'
import { RoomChart } from '../roomChart'
import '@testing-library/jest-dom'
import dayjs from 'dayjs'

import { enableFetchMocks } from 'jest-fetch-mock'

test('RoomChart', async () => {
  enableFetchMocks()
  fetchMock.mockResponseOnce(
    JSON.stringify([
      { value: 0, updatedAt: dayjs(new Date(2020, 7, 31, 0, 0, 0, 0)) }
    ]))
  const start = dayjs(new Date(2020, 7, 31, 0, 0, 0, 0))
  const end = dayjs(new Date(2020, 7, 31, 10, 0, 0, 0))

  const { container, getByRole } = render(<RoomChart roomName={"test"} logType={"test"} start={start} end={end} />)
  expect(fetchMock).toBeCalledWith(
    `/api/rooms/test/logs/test?start=${start.toISOString()}&end=${end.toISOString()}`)
  await waitForElementToBeRemoved(getByRole("progressbar"))
  expect(container).toMatchSnapshot()
})
