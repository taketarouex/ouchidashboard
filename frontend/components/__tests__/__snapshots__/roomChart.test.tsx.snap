// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomChart 1`] = `
<div>
  <div
    class="recharts-responsive-container"
    style="width: 90%; height: 400px;"
  >
    <div
      style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
    />
  </div>
</div>
`;
